pkgs/applications/science/math/pynac/default.nix

```diff
@@ -4,8 +4,9 @@
 , pkgconfig
 , flint
 , gmp
-, python2
+, python3
 , singular
```

Comment:
If this is no longer a dependency, it should be removed here as well. The dependency change seems odd to me though. Does the build fail without `ncurses`?

Reply:
It was removed in buildInputs because it was there twice. Singular is still needed for pynac.

Reply:
Makes sense. And what about `ncurses`? How does that relate to the python3 switch?

Reply:
Here is the difference the important difference between the configure log of pynac.

python3
----------
```
configure:4841: checking python extra libraries
configure:4848: result: -lpthread -ldl -lcrypt -lncurses -lutil -lm -lm
configure:4855: checking python extra linking flags
configure:4862: result: -Xlinker -export-dynamic
configure:4869: checking consistency of all components of python development environment
configure:4895: gcc -o conftest -g -O2  -I/nix/store/18656kvqazm74bj7k3mdkwmdlqfyf581-python3-3.8.6/include/python3.8  conftest.c  -L/nix/store/18656kvqazm74bj7k3mdkwmdlqfyf581-python3-3.8.6/lib -lpython3.8 -Xlinker -export-dynamic -lpthread -ldl -lcrypt -lncurses -lutil -lm -lm >&5
/nix/store/79rshjqnibwcx468m4i31wnmasaj6nf4-binutils-2.31.1/bin/ld: cannot find -lncurses
```
python2
----------
```
configure:4841: checking python extra libraries
configure:4848: result: -lpthread -ldl  -lutil -lm
configure:4855: checking python extra linking flags
configure:4862: result: -Xlinker -export-dynamic
configure:4869: checking consistency of all components of python development environment
configure:4895: gcc -o conftest -g -O2  -I/nix/store/xdha1qdqlh9hhymxq3w79zvjmwdyxqyz-python-2.7.18/include/python2.7  conftest.c  -L/nix/store/xdha1qdqlh9hhymxq3w79zvjmwdyxqyz-python-2.7.18/lib -lpython2.7 -Xlinker -export-dynamic -lpthread -ldl  -lutil -lm >&5
```
You can see on line 4848 that python contain more libraries than python2. If I understand correctly, this list may be all the libraries that python has been built with that are needed if you want to embed or link with python. That would explain with we need to add ncurses. The crypt library is also added but I guess that is it included in the standard environment.

I don't know enough about the build process to know if it would possible to remove the dependency on ncurses. And the problem may be deeper in the package of python3 than I taught initially.

You can confirm my finding by running `nix-shell -p python38 --run "python3.8-config --libs"` which should also show you the ncurses library.

I propose that we leave ncurses as a dependency. Maybe open an issue if we'd like to remove it in the future.

Reply:
Thanks for investigating! This seems very odd, but since it seems to be necessary for the build I agree that we can just leave it.